src/wp-content/themes/twentytwentyone/assets/css/ie.css

```diff
@@ -7720,79 +7564,927 @@ h1.page-title {
 	background-color: #000;
 }
 
-.has-gray-background-color[class] {
-	background-color: #39414d;
+.has-black-background-color[class] [class*=__inner-container] {
+	background-color: #000;
 }
 
 .has-dark-gray-background-color[class] {
 	background-color: #28303d;
 }
 
-.has-green-background-color[class] {
-	background-color: #d1e4dd;
+.has-dark-gray-background-color[class] [class*=__inner-container] {
+	background-color: #28303d;
 }
 
-.has-blue-background-color[class] {
-	background-color: #d1dfe4;
+.has-gray-background-color[class] {
+	background-color: #39414d;
 }
 
-.has-purple-background-color[class] {
-	background-color: #d1d1e4;
+.has-gray-background-color[class] [class*=__inner-container] {
+	background-color: #39414d;
 }
 
-.has-red-background-color[class] {
-	background-color: #e4d1d1;
+.has-light-gray-background-color[class] {
+	background-color: #f0f0f0;
 }
 
-.has-orange-background-color[class] {
+.has-light-gray-background-color[class] [class*=__inner-container] {
+	background-color: #f0f0f0;
+}
+
+.has-green-background-color[class] {
+	background-color: #d1e4dd;
+}
+
+.has-green-background-color[class] [class*=__inner-container] {
+	background-color: #d1e4dd;
+}
+
+.has-blue-background-color[class] {
+	background-color: #d1dfe4;
+}
+
+.has-blue-background-color[class] [class*=__inner-container] {
+	background-color: #d1dfe4;
+}
+
+.has-purple-background-color[class] {
+	background-color: #d1d1e4;
+}
+
+.has-purple-background-color[class] [class*=__inner-container] {
+	background-color: #d1d1e4;
+}
+
+.has-red-background-color[class] {
+	background-color: #e4d1d1;
+}
+
+.has-red-background-color[class] [class*=__inner-container] {
+	background-color: #e4d1d1;
+}
+
+.has-orange-background-color[class] {
+	background-color: #e4dad1;
+}
+
+.has-orange-background-color[class] [class*=__inner-container] {
 	background-color: #e4dad1;
 }
 
 .has-yellow-background-color[class] {
 	background-color: #eeeadd;
 }
 
+.has-yellow-background-color[class] [class*=__inner-container] {
+	background-color: #eeeadd;
+}
+
 .has-white-background-color[class] {
 	background-color: #fff;
 }
 
-:not(.has-text-color).has-black-background-color[class] {
-	color: #fff;
+.has-white-background-color[class] [class*=__inner-container] {
+	background-color: #fff;
 }
 
-:not(.has-text-color).has-gray-background-color[class] {
-	color: #fff;
+.has-background:not(.has-text-color).has-black-background-color[class] [class*=__inner-container] {
+	color: #28303d;
 }
 
-:not(.has-text-color).has-dark-gray-background-color[class] {
-	color: #fff;
+.has-background:not(.has-text-color).has-black-background-color[class] [class*=__inner-container] {
```

Comment:
@allancole Why are these repeated 20 times? This can't be right?

Reply:
Looks like a bug in postcss-css-variables when using `:not` and local variables - I can report this upstream, but I think we'll need a workaround in the meantime.

Reply:
I’m assuming it would be _bad form_ to just remove the duplicates from the `ie.css` stylesheet manually. Maybe we could add [something like this](https://developer.aliyun.com/mirror/npm/package/postcss-discard-duplicates) or [similar](https://github.com/ChristianMurphy/postcss-combine-duplicated-selectors) to the build process as a workaround?

Reply:
I was just coming here to suggest that we could try that.

Reply:
It looks like `postcss-discard-duplicates` should work, I'm trying it out now 👍

Reply:
Created a ticket & PR for `postcss-discard-duplicates`, which highlighted that we _already_ have some duplicate CSS from a postcss-css-variables bug. https://github.com/WordPress/wordpress-develop/pull/835